### ISSUE TITLE
Global flags are in the way

### ISSUE THREAD

**Issue by philderbeast** [haskell/cabal#9783]
Lots of commands take targets.

_Originally posted by @philderbeast in https://github.com/haskell/cabal/issues/9744#issuecomment-1980928890_

I thought of maybe listing the commands that take targets but quickly gave up, not wanting to scroll up past 6 pages of global flags to see whether the command takes `[TARGETS]`;

```
$ cabal build --help
Compile targets within the project.

Usage: cabal build [TARGETS] [FLAGS]

Description ...

Flags for build (includes global flags):
...
```

Can we not show the global flags by default?

**Comment by geekosaur** [comment]
A common pattern is to list the global flags in their own section.

**Comment by philderbeast** [comment]
Yes stack has a separate command for raising the living dead of global flags;

> Command 'stack --help' for global options that apply to all subcommands.
